src/main/app/src/utils/hakukohde/hakukohdeRemoteErrorsToFormErrors.ts

```diff
@@ -34,5 +34,20 @@ export const hakukohdeRemoteErrorsToFormErrors: RemoteErrorsToFormErrors = ({
         ];
       }
     }
+  } else if (errorType === 'missingMsg' && /painotetutArvosanat/.test(path)) {
+    const fieldName = path.replace('metadata.', '');
+    let errorObject = {
+      field: fieldName,
+      errorKey: 'validointivirheet.pakollinen',
+    };
+
+    if (/koodiUrit/.test(fieldName)) {
+      errorObject = {
+        ...errorObject,
+        field: fieldName.replace('koodiUrit', 'painotettuOppiaine'),
+      };
+    }
```

Comment:
Eiks tän voi tehdä vaan näin?
```suggestion
    return {
      field: path.replace('metadata.', '').replace('koodiUrit', 'painotettuOppiaine');,
      errorKey: 'validointivirheet.pakollinen',
    }
```